newIDE/app/src/ProjectManager/ExtensionsPropertiesDialog.js

```diff
@@ -0,0 +1,110 @@
+// @flow
+import { Trans } from '@lingui/macro';
+
+import * as React from 'react';
+import FlatButton from '../UI/FlatButton';
+import Dialog from '../UI/Dialog';
+import Text from '../UI/Text';
+import { ColumnStackLayout } from '../UI/Layout';
+import EmptyMessage from '../UI/EmptyMessage';
+import PropertiesEditor from '../PropertiesEditor';
+import propertiesMapToSchema from '../PropertiesEditor/PropertiesMapToSchema';
+import { List } from '@material-ui/core';
+
+type Props = {|
+  project: gdProject,
+  open: boolean,
+  onApply: Function,
+|};
+
+class ExtensionsPropertiesDialog extends React.Component<Props> {
+  _onApply = () => {
+    this.props.onApply();
+  };
+
+  _mapExtensionValueToPropertyDescriptors(
+    properties: gdMapStringPropertyDescriptor,
+    extensionName: string,
+    project: gdProject
+  ) {
+    const keys = properties.keys().toJSArray();
+    for (let key of keys) {
+      properties
+        .get(key)
+        .setValue(
+          project.getExtensionPropertiesManager().getValue(extensionName, key)
+        );
+    }
+    return properties;
+  }
+
+  render() {
+    const { project } = this.props;
+    const allExtensions = project
+      .getCurrentPlatform()
+      .getAllPlatformExtensions();
+    const properties = [];
+    for (let i = 0; i < allExtensions.size(); i++) {
+      const extension = allExtensions.at(i);
+      const propertiesSchema = propertiesMapToSchema(
+        extension.getAllProperties(),
+        instance =>
+          this._mapExtensionValueToPropertyDescriptors(
+            extension.getAllProperties(),
+            extension.getName(),
+            project
+          ),
```

Comment:
This looks very complex. Why can't this just be `project.getExtensionPropertiesManager().getvalue(extension.getName(), propertyName)`?

Reply:
I thought I had to pass a propertyDescriptor, but `project.getExtensionPropertiesManager().getvalue(extension.getName(), propertyName)` would just be the value.

Reply:
You're right my bad! Let me think about it. `_mapExtensionValueToPropertyDescriptors` is not going to work because you're modifying in memory the properties of the extension.
We might have to make a static method in C++, which takes an extension, a project, and returns the properties with the values set.

In other words, what you did in JS, but do it as a method in gd::Project... or better in ExtensionPropertiesManager :) 
It's a bit cumbersome sorry - this was originally designed for objects that are returning their properties while here we have properties in extension and objects storing them.